### PR TITLE
Handle detection of None in the to field of the transaction for coinbase plugin as a part of filtering

### PR DESCRIPTION
**Pull request by kiates**
Possible solution for #131 ??

### CODE REVIEW

**Comment by eprbell** [comment]
> Possible solution for #131 ??

To understand if this is the right fix we need to look at the raw data from Coinbase for the culprit transaction. Here's how to get that information:
* `rm -rf log/`
* edit your .ini file and make sure the Coinbase section has `thread_count = 1`
* rerun DaLI with debug logging: `LOG_LEVEL=DEBUG bin/dali_us -t 1 <rest of your parameters>`
* open the log that has just been generated in `log/` and go to the bottom
* look for the last line that begins with: `<date>/Coinbase/<user>/DEBUG:` and paste it here (feel free to remove personal information)

**Comment by kiates** [comment]
Here is the full transaction:

> {
	'id': 'redacted',
	'type': 'send',
	'status': 'completed',
	'amount': {	'amount': '-2.833888', 'currency': 'ADA' },
	'native_amount': {'amount': '-1.02', 'currency': 'USD' },
	'description': None,
	'created_at': '2023-03-24T23: 31: 59Z',
	'updated_at': '2023-03-24T23: 31: 59Z',
	'resource': 'transaction',
	'resource_path': '/v2/accounts/redacted',
	'instant_exchange': False,
	'off_chain_status': 'completed',
	'network': {
		'status': 'off_blockchain',
		'status_description': None
    },
	'to': None, 'details': {
		'title': None,
		'subtitle': None,
		'header': None,
		'health': 'positive'
    },
	'hide_native_amount': False
}

**Comment by eprbell** [comment]
> Here is the full transaction:
> 
> > {
> > 'id': 'redacted',
> > 'type': 'send',
> > 'status': 'completed',
> > 'amount': {	'amount': '-2.833888', 'currency': 'ADA' },
> > 'native_amount': {'amount': '-1.02', 'currency': 'USD' },
> > 'description': None,
> > 'created_at': '2023-03-24T23: 31: 59Z',
> > 'updated_at': '2023-03-24T23: 31: 59Z',
> > 'resource': 'transaction',
> > 'resource_path': '/v2/accounts/redacted',
> > 'instant_exchange': False,
> > 'off_chain_status': 'completed',
> > 'network': {
> > 'status': 'off_blockchain',
> > 'status_description': None
> > },
> > 'to': None, 'details': {
> > 'title': None,
> > 'subtitle': None,
> > 'header': None,
> > 'health': 'positive'
> > },
> > 'hide_native_amount': False
> > }

Oh, wow, the `to` field is None and the fields of `details` are almost all `None`: never saw this before. Could you describe what kind of transaction this is? Really curious what might generate such an output.

**Comment by kiates** [comment]
It looks like it may be a transaction to move coin into their staking system.

In the regular (Primary blance) area it shows what looks like the original transaction to stake it:
image

Then in staking interface (Earning), shows staked ADA:
image

File: src/dali/plugin/input/rest/coinbase.py [hunk 1]
```diff
@@ -563,6 +563,8 @@ def _process_transfer(
             if amount < ZERO:
                 if (
```
Comment (eprbell) [inline src/dali/plugin/input/rest/coinbase.py:564]: Disable pylint warning with `# pylint: disable=too-many-boolean-expressions`

**Review by eprbell — CHANGES_REQUESTED** [review]
Thanks for the context! The fix looks good (just suppress the pylint warning).

File: src/dali/plugin/input/rest/coinbase.py [hunk 2]
```diff
@@ -561,8 +561,8 @@ def _process_transfer(
             transaction_network = transaction[_NETWORK]
             crypto_hash: str = transaction_network[_HASH] if _HASH in transaction_network else Keyword.UNKNOWN.value
             if amount < ZERO:
-                if (
-                    _TO in transaction
+                if ( # pylint: disable=too-many-boolean-expressions
```
Comment (kiates) [inline src/dali/plugin/input/rest/coinbase.py:564]: Like this?  Not sure the rules for where this goes, but it seems like it is on the first line of an if.

**Review by eprbell — APPROVED** [review]
LGTM!

**Pull request merged**